src/main/java/org/opensearch/sdk/SDKClient.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.io.IOException;
+
+import org.apache.http.HttpHost;
+
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestClientBuilder;
+import org.opensearch.client.json.jackson.JacksonJsonpMapper;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.transport.OpenSearchTransport;
+import org.opensearch.client.transport.rest_client.RestClientTransport;
+
+/**
+ * This class creates a Client for SDK to make requests to OpenSearch
+ */
+public class SDKClient {
+    private final Logger logger = LogManager.getLogger(SDKClient.class);
+    private OpenSearchClient client;
+    private RestClient restClient = null;
+
+    /**
+     * Creates client for SDK
+     * @param hostAddress The address client can connect to
+     * @param port The port of the address
+     * @throws IOException if client failed
+     * @return SDKClient
+     */
+    public OpenSearchClient createClient(String hostAddress, int port) throws IOException {
+        RestClientBuilder builder = RestClient.builder(new HttpHost(hostAddress, port));
+        builder.setStrictDeprecationMode(true);
+        builder.setHttpClientConfigCallback(httpClientBuilder -> {
+            final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            // credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+            try {
+                return httpClientBuilder.setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                    .setSSLContext(SSLContextBuilder.create().loadTrustMaterial(null, (chains, authType) -> true).build());
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        restClient = builder.build();
+
+        // Create Client
+        OpenSearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
+        client = new OpenSearchClient(transport);
+        return client;
+    }
+
+    /**
+     *
+     * @throws IOException if closing the client fails
+     */
+    public void doClose() throws IOException {
```

Comment:
Tracing the call to `restClient.close()` it leads to `CloseableHttpAsyncClientBase` where any `IOException` is caught. So you can (and should) remove this `throws` clause so callers don't need to waste time handling an exception that won't occur.  (@saratvemulapalli please confirm this suggestion.)
https://jar-download.com/artifacts/org.apache.httpcomponents/httpasyncclient/4.1.4/source-code/org/apache/http/impl/nio/client/CloseableHttpAsyncClientBase.java

Reply:
`restClient.close()` needs an exception otherwise it gives an error. If `IOException` doesn't look here we can change it to `Exception`.

Reply:
Not sure how you've traced it @dbwiddis. 
I see this goes to:

https://github.com/opensearch-project/OpenSearch/blob/3a97d4cad65fa19126340aabf7e8bd2b1ce2b822/client/rest/src/main/java/org/opensearch/client/RestClient.java#L587

Which does throw IOException. Either we have to handle it or throw it back to the caller.

Reply:
Yes, `RestClient` has `throws` in the signature because the `client` in the call is based on an interface.  However, the specific implementation of that interface does not throw an exception. That was my original.

However, I retract my comment as we should not rely on implementation details and thus it is correct to treat `restClient.close()` as possibly throwing an exception.   (I had some doubt which was why I asked for a second opinion here).